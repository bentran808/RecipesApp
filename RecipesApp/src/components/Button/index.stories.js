import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react-native';
import React from 'react';
import CenterView from 'storybook/stories/CenterView';
import Button from '.';

const handlePress = action('clicked-button');
storiesOf('Button', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('default', () => <Button testID="testID" title="Hello Button" onPress={handlePress} />)
  .add('bold text', () => (
    <Button testID="testID" title="Hello Button" bold onPress={handlePress} />
  ))
  .add('type outlined', () => (
    <Button testID="testID" title="Hello Button" type="outlined" onPress={handlePress} />
  ))
  .add('type contained', () => (
    <Button testID="testID" title="Hello Button" type="contained" onPress={handlePress} />
  ));
