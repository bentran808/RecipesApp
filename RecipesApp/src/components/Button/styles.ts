import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 20,
    marginHorizontal: 10,
    borderRadius: 100,
    borderWidth: 1,
    justifyContent: 'center',
    alignItems: 'center'
  }
});

export default styles;
