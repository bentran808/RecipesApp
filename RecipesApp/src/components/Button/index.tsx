import React from 'react';
import isEqual from 'react-fast-compare';
import { Text, TouchableOpacity, View } from 'react-native';
import styles from './styles';

interface Props {
  testID: string;
  title: string;
  onPress: () => void;
  type?: string;
  borderColor?: string;
  paddingVertical?: number;
  paddingHorizontal?: number;
  fontSize?: number;
  color?: string;
  bold?: boolean;
  style?: any;
  disabled?: boolean;
}

const Button = ({
  testID,
  title,
  onPress,
  type = 'text', // 'text' | 'outlined' | 'contained'
  borderColor = '#2cd18a',
  paddingVertical = 10,
  paddingHorizontal = 60,
  fontSize = 14,
  color = '#2cd18a',
  bold = false,
  style,
  disabled = false
}: Props) => {
  const fontWeight = bold ? 'bold' : 'normal';
  let textColor = disabled ? 'gray' : color;
  let additionalStyles;

  switch (type) {
    case 'outlined':
      additionalStyles = {
        borderWidth: 1,
        borderColor,
        paddingVertical,
        paddingHorizontal
      };
      break;
    case 'contained':
      additionalStyles = {
        backgroundColor: disabled ? 'gray' : color,
        paddingVertical,
        paddingHorizontal
      };
      textColor = 'white';
      break;

    default:
      additionalStyles = {};
      break;
  }

  return (
    <TouchableOpacity disabled={disabled} testID={testID} onPress={onPress}>
      <View style={[styles.btn, additionalStyles, style]}>
        <Text style={[{ fontSize, color: textColor, fontWeight, textAlign: 'center' }]}>
          {title}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

export default React.memo(Button, isEqual);
