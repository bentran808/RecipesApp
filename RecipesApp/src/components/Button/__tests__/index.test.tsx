import Button from 'components/Button';
import renderer from 'react-test-renderer';

describe('Button Component', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(<Button testID="testID" title="Test" onPress={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render contained button correctly', () => {
    const tree = renderer
      .create(<Button testID="testID" title="Test" type="contained" onPress={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
