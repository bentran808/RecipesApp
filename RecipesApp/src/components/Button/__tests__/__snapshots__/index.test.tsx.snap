// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render contained button correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="testID"
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 100,
          "marginHorizontal": 10,
        },
        Object {
          "backgroundColor": "#2cd18a",
          "paddingHorizontal": 60,
          "paddingVertical": 10,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "normal",
          },
        ]
      }
    >
      Test
    </Text>
  </View>
</View>
`;

exports[`Button Component should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="testID"
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 100,
          "marginHorizontal": 10,
        },
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2cd18a",
            "fontSize": 14,
            "fontWeight": "normal",
          },
        ]
      }
    >
      Test
    </Text>
  </View>
</View>
`;
